module-apps/application-notes/windows/NoteEditWindow.cpp

```diff
@@ -150,6 +151,24 @@ namespace app::notes
                     std::make_unique<gui::OptionsWindowOptions>(noteEditOptions(application, *notesRecord, edit)));
             }
         }
+        if (isCurrentTextDifferentThanSaved() &&
+            (inputEvent.isShortRelease(gui::KeyCode::KEY_RF) || inputEvent.isLongRelease(gui::KeyCode::KEY_RF))) {
+
+            // Show a popup warning about possible data loss
+            auto metaData = std::make_unique<gui::DialogMetadataMessage>(
+                gui::DialogMetadata{utils::translate("unsaved_changes"),
+                                    "delete_128px_W_G",
+                                    utils::translate("exit_without_saving"),
+                                    "",
+                                    [=]() -> bool {
+                                        application->returnToPreviousWindow(); // To exit this popup
+                                        application->returnToPreviousWindow(); // To switch back to previous window
```

Comment:
What about switching directly to some window instead of returning? What if some popup appear in the middle?

Reply:
Unfortunately is not that easy, because creating new note and editing note has different flow, I mean different window to return.

And if some other popup will appear in the middle then, IMO, is better to return to this other popup, and give the user the ability to see it, and the user will try to go back thru this "unsaved changes" popup again.

The tethering popup no longer allows user to switch the window to the home screen. Since: https://github.com/mudita/MuditaOS/pull/4578